packages/workflow/src/sinks.ts

```diff
@@ -44,14 +45,64 @@ export interface SinkCall {
 }
 
 /**
- * Sink interface for forwarding logs from the Workflow sandbox to the Worker
+ * Get a reference to Sinks for exporting data out of the Workflow.
+ *
+ * These Sinks **must** be registered with the Worker in order for this
+ * mechanism to work.
+ *
+ * @example
+ * ```ts
+ * import { proxySinks, Sinks } from '@temporalio/workflow';
+ *
+ * interface MySinks extends Sinks {
+ *   logger: {
+ *     info(message: string): void;
+ *     error(message: string): void;
+ *   };
+ * }
+ *
+ * const { logger } = proxySinks<MyDependencies>();
+ * logger.info('setting up');
+ *
+ * export function myWorkflow() {
+ *   return {
+ *     async execute() {
+ *       logger.info("hey ho");
+ *       logger.error("lets go");
+ *     }
+ *   };
+ * }
+ * ```
  */
-export interface LoggerSinks extends Sinks {
-  defaultWorkerLogger: {
-    trace(message: string, attrs?: Record<string, unknown>): void;
-    debug(message: string, attrs?: Record<string, unknown>): void;
-    info(message: string, attrs?: Record<string, unknown>): void;
-    warn(message: string, attrs?: Record<string, unknown>): void;
-    error(message: string, attrs?: Record<string, unknown>): void;
-  };
+export function proxySinks<T extends Sinks>(): T {
+  return new Proxy(
+    {},
+    {
+      get(_, ifaceName) {
+        return new Proxy(
+          {},
+          {
+            get(_, fnName) {
+              return (...args: any[]) => {
+                const activator = assertInWorkflowContext(
+                  'Proxied sinks functions may only be used from a Workflow Execution.'
+                );
+                activator.sinkCalls.push({
+                  ifaceName: ifaceName as string,
+                  fnName: fnName as string,
+                  // Sink function don't get called immediately. Make a clone of sink's args, so that further mutations
+                  // to these objects don't corrupt the args that the sink function will receive. Only available from node 17.
```

Comment:
```suggestion
                  // Sink function doesn't get called immediately. Make a clone of the sink's args, so that further mutations
                  // to these objects don't corrupt the args that the sink function will receive. Only available from node 17.
```